RabbitMq/Consumer.php

```diff
@@ -281,25 +307,36 @@ private function chooseWaitTimeout()
              * Respect the idle timeout if it's set and if it's less than
              * the remaining allowed execution.
              */
-            if (
-                $this->getIdleTimeout()
+            if ($this->getIdleTimeout()
                 && $this->getIdleTimeout() < $allowedExecutionSeconds
             ) {
-                return array(
-                    'timeoutType' => self::TIMEOUT_TYPE_IDLE,
-                    'seconds' => $this->getIdleTimeout(),
-                );
+                $waitTimeout = $this->getIdleTimeout();
+            } else {
+                $waitTimeout = $allowedExecutionSeconds;
             }
+        } else {
+            $waitTimeout = $this->getIdleTimeout();
+        }
 
-            return array(
-                'timeoutType' => self::TIMEOUT_TYPE_GRACEFUL_MAX_EXECUTION,
-                'seconds' => $allowedExecutionSeconds,
-            );
+        if ($this->getTimeoutWait()) {
+            $waitTimeout = min($waitTimeout, $this->getTimeoutWait());
         }
+        return $waitTimeout;
+    }
 
-        return array(
-            'timeoutType' => self::TIMEOUT_TYPE_IDLE,
-            'seconds' => $this->getIdleTimeout(),
-        );
+    /**
```

Comment:
Please remove the comment.

Reply:
Do you mean I should completely remove @param annotation?

Reply:
Yes. Once you've added typehint for parameter, comment it's useless